splunk_handler/__init__.py

```diff
@@ -38,27 +39,33 @@ def emit(self, record):
 
         thread.start()
 
-    def _init_client(self):
-
-        return client.connect(
-                   host=self.host,
-                   port=self.port,
-                   username=self.username,
-                   password=self.password)
-
     def _async_emit(self, record):
 
-        global _client
-
-        if not _client:
-            _client = self._init_client()
-
         try:
-            _client.indexes[self.index].submit(
-                 self.format(record),
-                 host=socket.gethostname(),
-                 source=record.pathname,
-                 sourcetype='json')
+
+            if self.source is None:
+                source = record.pathname
+            else:
+                source = self.source
+
+            params = {
+                'host': self.hostname,
+                'index': self.index,
+                'source': source,
+                'sourcetype': self.sourcetype
+            }
+            url = 'https://%s:%s/services/receivers/simple' % (self.host, self.port)
+            payload = self.format(record)
```

Comment:
The string literal before it's sent to splunk would look like:
`line one\\nline two\\nline three`